### PR TITLE
Add docker compose v2 to linux

### PR DESCRIPTION
**Pull request by donbobka**
As per discussion in https://github.com/buildkite/elastic-ci-stack-for-aws/issues/1022 docker compose v2 can coexist with v1 + `docker-compose-plugin` now supports [opt-in feature](https://github.com/buildkite-plugins/docker-compose-buildkite-plugin/pull/334) to enable docker-compose v2

### CODE REVIEW

**Comment by tetienne** [comment]
Any update here? The check is still pending.

**Comment by jesseduffield** [comment]
I believe github can get stuck sometimes. If you `git commit --amend --no-edit` and force push that may resolve the issue.

While I'm here I'll mention that I'm very keen on this PR being merged!

File: packer/linux/scripts/install-docker.sh [hunk 1]
```diff
@@ -4,6 +4,7 @@ set -eu -o pipefail
 DOCKER_VERSION=20.10.17
 DOCKER_RELEASE="stable"
 DOCKER_COMPOSE_VERSION=1.29.2
+DOCKER_COMPOSE_V2_VERSION=2.11.2
```
Comment (triarius) [inline packer/linux/scripts/install-docker.sh:7]:
```suggestion
DOCKER_COMPOSE_V2_VERSION=2.12.2
```

**Review by triarius — APPROVED** [review]
@donbobka Thanks for your PR! Not sure what happened with the webhook, so copied your changes into a branch and sheparded it through buildkite's CI: https://github.com/buildkite/elastic-ci-stack-for-aws/compare/master...triarius/test-dcv2

Long and short of it is it looks good 🪅 Please accept the version bump and we will merge.

**Comment by jesseduffield** [comment]
@triarius Question for you: once this is merged, how soon after could we expect a new release containing the change?

**Comment by triarius** [comment]
@jesseduffield It's out now: https://github.com/buildkite/elastic-ci-stack-for-aws/releases/tag/v5.12.0It's done: https://github.com/buildkite/elastic-ci-stack-for-aws/releases/tag/v5.12.0.

**Pull request merged**